src/assign.c

```diff
@@ -930,48 +929,55 @@ const char *memrecycle(SEXP target, SEXP where, int start, int len, SEXP source,
 
 #define COERCE_ERROR(targetType) error("type '%s' cannot be coerced to '%s'", type2char(TYPEOF(source)), targetType); // 'targetType' for integer64 vs double
 
-  const int off = (length(where)?0:start); // off = target offset; e.g. called from rbindlist with where=R_NilValue and start!=0
-  const bool mem = length(where)==0 && slen==len;  // only used if types match too
+  const int off = length(where) ? 0 : start;  // off = target offset; e.g. called from rbindlist with where=R_NilValue and start!=0
+  const bool mc = length(where)==0 && slen==len;  // mc=memcpy; only used if types match too
+  const int *wd = length(where) ? INTEGER(where)+start : NULL;
   switch (TYPEOF(target)) {
   case RAWSXP: {
     Rbyte *td = RAW(target) + off;
     switch (TYPEOF(source)) {
-    case RAWSXP:  if (mem) {
+    case RAWSXP:
+      if (mc) {
                     memcpy(td, RAW(source), slen*sizeof(Rbyte)); break;
-          } else    BODY(Rbyte, RAW,    Rbyte, val,                                     td[i]=cval)
+      } else        BODY(Rbyte, RAW,    Rbyte, val,                                     td[i]=cval)
```

Comment:
Newline for BODY expression would be useful for coverage

Reply:
Please see image I pasted here before which shows the newline isn't needed for `else`.
https://github.com/Rdatatable/data.table/pull/3909#issuecomment-535728419